src/config.py

```diff
@@ -40,6 +40,12 @@ def get_logs_dir(self):
     def get_repos_dir(self):
         return self.config["STORAGE"]["REPOS_DIR"]
     
+    def get_openai_base_url(self):
+        return self.config["API_ENDPOINTS"]["OPENAI_BASE_URL"]
+    
+    def get_search_engine(self):
+        return self.config["USER_PREFS"]["SEARCH_ENGINE"]
+    
     def set_bing_api_key(self, key):
         self.config["API_KEYS"]["BING"] = key
         self.save_config()
```

Comment:
**Optimization Issue**: The methods get_openai_base_url and get_search_engine introduce new configuration accessors without corresponding setters or validation. This can lead to runtime errors if the configuration keys are missing or misspelled. <br> **Fix**: Implement validation checks in the configuration loading phase to ensure all required keys are present and correctly formatted. Additionally, consider providing setter methods or a more dynamic way to update these configurations to ensure consistency and error handling. <br> **Code Suggestion**: 
 ```
 +class ConfigManagement:
 +    @staticmethod
 +    def get_openai_base_url(config):
 +        # Implement validation checks here
 +        return config["API_ENDPOINTS"]["OPENAI_BASE_URL"]
 +
 +    @staticmethod
 +    def get_search_engine(config):
 +        # Implement validation checks here
 +        return config["USER_PREFS"]["SEARCH_ENGINE"]
 ```

Reply:
[its-arun](https://github.com/its-arun) ["USER_PREFS"]["SEARCH_ENGINE"] is not included in config.toml

Reply:
> [its-arun](https://github.com/its-arun) ["USER_PREFS"]["SEARCH_ENGINE"] is not included in config.toml

It is included with this commit https://github.com/stitionai/devika/blob/d80eb55a576873cefc18b615e5c9b80b28ae7ae8/config.toml#L20